Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,337 @@
+
+        LOG.debug("end transformResponseToAuditMsg() -- NHIN");
+
+        return result;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeResponse msg, AssertionType assertion) {
+        if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+            LOG.error("One of more UserInfo fields from the Assertion object were null.");
+            return true;
+        }
+
+        if (areRequiredRequestFieldsNull(msg)) {
+            LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeRequest msg, AssertionType assertion) {
+        if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+            LOG.error("One of more UserInfo fields from the Assertion object were null.");
+            return true;
+        }
+
+        if (areRequiredRequestFieldsNull(msg)) {
+            LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredRequestFieldsNull(COREEnvelopeRealTimeRequest msg) {
+        if (NullChecker.isNullish(msg.getPayloadType())) {
+            LOG.error("CORE X12 Realtime PayloadType is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getProcessingMode())) {
+            LOG.error("CORE X12 Realtime ProcessingMode is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getPayloadID())) {
+            LOG.error("CORE X12 Realtime PayloadID is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getTimeStamp())) {
+            LOG.error("CORE X12 Realtime TimeStamp is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getSenderID())) {
+            LOG.error("CORE X12 Realtime SenderID is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getReceiverID())) {
+            LOG.error("CORE X12 Realtime ReceiverID is empty...");
+            return true;
+        }
+
+        if (NullChecker.isNullish(msg.getCORERuleVersion())) {
+            LOG.error("CORE X12 Realtime CORERuleVersion is empty...");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredRequestFieldsNull(COREEnvelopeRealTimeResponse msg) {
+        if (msg.getPayloadType() == null || msg.getPayloadType().length() <= 0) {
+            LOG.error("CORE X12 Realtime PayloadType is empty...");
+            return true;
+        }
+
+        if (msg.getProcessingMode() == null || msg.getProcessingMode().length() <= 0) {
+            LOG.error("CORE X12 Realtime ProcessingMode is empty...");
+            return true;
+        }
+
+        if (msg.getPayloadID() == null || msg.getPayloadID().length() <= 0) {
```

Comment:
As specified by Jason, these checks can be replaced by isNullish(), it checks both for null & empty string. Fix it for the below fields also timeStamp, senderID, receiverID and CORERuleVersion